src/averageAges.js

```diff
@@ -37,7 +54,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let females = people.filter(({ sex }) => sex === FEMALE_SIGN);
+
+  if (withChildren) {
+    females = females.filter(({ name }) => {
+      return people.some(({ mother }) => mother === name);
+    });
+  }
+
+  const ages = females.map(({ born, died }) => died - born);
```

Comment:
You do a similar calculation twice, you should move it to the middle of the `averageOfArray` function and to the middle of `reduce`, to avoid traversing the array twice when it can be done once.

Reply:
Sure, I can move that calculation to some function, but I won`t agree that I should move it exactly to `averageOfArray`, because i call `averageOfArray` inside of `calculateAverageAgeDiff` function, while this calculation is not used in `calculateAverageAgeDiff`.